devel/README.md

```diff
@@ -1,7 +1,341 @@
+  - [Enable pre push hook](#enable-pre-push-hook)
+  - [Create a branch and make changes](#create-a-branch-and-make-changes)
+  - [Keeping your fork in sync](#keeping-your-fork-in-sync)
+  - [Committing changes to your fork](#committing-changes-to-your-fork)
+  - [Creating a pull request](#creating-a-pull-request)
+  - [Getting a code review](#getting-a-code-review)
+  - [When to retain commits and when to squash](#when-to-retain-commits-and-when-to-squash)
+- [Using the code base](#using-the-code-base)
+  - [Building the code](#building-the-code)
+  - [Cleaning outputs](#cleaning-outputs)
+  - [Running tests](#running-tests)
+  - [Getting coverage numbers](#getting-coverage-numbers)
+  - [Auto-formatting source code](#auto-formatting-source-code)
+  - [Running the linters](#running-the-linters)
+  - [Running race detection tests](#running-race-detection-tests)
+  - [Adding dependencies](#adding-dependencies)
+  - [About testing](#about-testing)
+- [Local development scripts](#collection-of-scripts-and-notes-for-developing-for-istio)
+- [MacOS tips](#macos-tips)
+
+Other docs you should look at:
+
+- Each components additional development docs like [Mixer](https://github.com/istio/mixer/tree/master/doc/dev/development.md)'s
+- [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
+- [Go style mistakes](https://github.com/golang/go/wiki/CodeReviewComments)
+
+## Prerequisites
+
+Istio components code bases has only few external dependencies you
+need to setup before being able to build and run the code.
+
+### Setting up Go
+
+Mixer for instance is written in the [Go](http://golang.org) programming language.
+To build Mixer, you'll need a Go development environment. Builds for
+Mixer require Go version 1.8. If you haven't set up a Go development
+environment, please follow [these instructions](http://golang.org/doc/code.html)
+to install the Go tools.
+
+Set up your GOPATH and add a path entry for Go binaries to your PATH. Typically
+added to your ~/.profile:
+
+```shell
+export GOPATH=~/go
+export PATH=$PATH:$GOPATH/bin
+```
+
+### Setting up Bazel
+
+Istio components are built using the bazel build system. See
+[here](https://bazel.build/versions/master/docs/install.html) for the
+installation procedures.
+
+### Setting up Docker
+
+To run some of Istio's examples and tests, you need to set up Docker server.
+Please follow [these instructions](https://docs.docker.com/engine/installation/)
+for how to do this for your platform.
+
+### Setting up personal access token
+
+This is only necessary for core contributors / to push changes to the main repos. 
+You can make Pull Requests without it but the additional security is recommended for everyone.
+
+To be part of the Istio organization, we require two-factor authentication, and
+you must setup a personal access token to enable push via HTTPS. Please follow [these
+instructions](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
```

Comment:
you can't split the url but you can split the text of the link (it works, also not changed from earlier doc)